pkg/process/config/yaml_config.go

```diff
@@ -47,6 +47,7 @@ func (a *AgentConfig) loadSysProbeYamlConfig(path string) error {
 
 	if config.Datadog.GetBool(key(spNS, "enabled")) {
 		a.EnabledChecks = append(a.EnabledChecks, "connections")
+		a.Enabled = true
```

Comment:
Might be useful to emit a log message if we detect that it was false before this, and that we're changing this to enabled:

e.g.
```
INFO enabling process-agent for connections check as the system-probe is enabled...
```